design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,178 @@
+| Host Annotation  | Defines the checks that we should do on authentication request for host. In JWT authentication its a claim check. |
+| Mandatory Claims | JWT-specific claims that a host must validate their existance and value |
+| Claims Mapping   | Mapping between claim name and name of host annotation       |
+
+## Useful Links
+[//]: # "Add links that may be useful for the reader"
+
+| **Name**   | **Link**                                                     |
+| ---------- | ------------------------------------------------------------ |
+| JWT Claims | https://auth0.com/docs/tokens/json-web-tokens/json-web-token-claims |
+
+## Requirements
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+1. Required fields
+   1. In the authenticator policy, the user will have the ability to declare which claims must be in host annotations. As a result, also in the JWT token
+   2. When sending an authentication request, the authenticator MUST validate those claims - existence and content 
+2. Mapping 
+   1. The user should have the ability to map a given "technical" claim name to a more familiar / business one - i.e, in git-lab. Instead of mentioning in the policy "sub" claim, the user will have the ability to mention "job"
+
+## Solution
+User will define the token schema variables in the authenticator policy. The variables will be checked in the JWT authentication request.
+
+* The usage of this variables is optional and non mandatory. User can authenticate to JWT in the same way he authenticated in stage 1. Using the feature adds more security checks to the authentication.
+
+### User Interface
+
+#### Authenticator Policy
+
+Two new variables will be added to the authenticator policy:
+
+* mandatory_claims
+* claims_mapping
+
+```yaml
+    - !policy
+    id: conjur/authn-jwt/VendorY
+    body:
+    - !webservice
+    
+    - !variable
+      id: mandatory_claims
+      
+    - !variable
+      id: claims_mapping
+```
+
+#### Variable Values Example
+
+| Name             | Value                   | Description                                      |
+| ---------------- | ----------------------- | ------------------------------------------------ |
+| mandatory_claims | branch, job, aud        | List of claims as they appear in host annotation |
+| claims_mapping   | {ref: branch, sub: job} | Mapping between host annotation to JWT claim     |
+
+* For any claim mapped, we will add the host annotation to the `mandatory_claims` variable and not the JWT claim name.
+* The values above are examples and they can be any claims.
+
+
+## Design
+[//]: # "Add any diagrams, charts and explanations about the design aspect of the solution. Elaborate also about the expected user experience for the feature"
+
+The following class diagram, represent the flow of checking the mandatory claims and the claim mapping.
+
+### Class Diagrams
+
+
+![image-20210711171238701](token-schema-class-diagram.png)
+
+### Flow Explanation
```

Comment:
What will be the flow for mapping that is not mandatory?
What happen to the flow if the claim doesn't exists or not familiar (as it is host claim - not jwt claim)?